features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/opentelemetry/OtelContextTest.kt

```diff
@@ -0,0 +1,204 @@
+    @Mock
+    private lateinit var mockRootSpan: Span
+
+    private lateinit var testedContext: OtelContext
+
+    @BeforeEach
+    fun setUp() {
+        mockContext = mock()
+        testedContext = OtelContext(mockContext, mockCurrentSpan, mockRootSpan)
+    }
+
+    // region get by key
+
+    @Test
+    fun `M return current span W get { key is OTEL_CONTEXT_SPAN_KEY }`() {
+        // Given
+        val key = ContextKey.named<Span>(OtelContext.OTEL_CONTEXT_SPAN_KEY)
+
+        // When
+        val result = testedContext.get(key)
+
+        // Then
+        assertThat(result).isEqualTo(mockCurrentSpan)
+    }
+
+    @Test
+    fun `M return root span W get { key is OTEL_CONTEXT_ROOT_SPAN_KEY }`() {
+        // Given
+        val key = ContextKey.named<Span>(OtelContext.OTEL_CONTEXT_ROOT_SPAN_KEY)
+
+        // When
+        val result = testedContext.get(key)
+
+        // Then
+        assertThat(result).isEqualTo(mockRootSpan)
+    }
+
+    @Test
+    fun `M delegate to wrapped Context W get { key is not recognized }`(forge: Forge) {
+        // Given
+        val fakeKeyName = forge.anAlphabeticalString()
+        val key = ContextKey.named<String>(fakeKeyName)
+        val fakeKeyValue = forge.anAlphabeticalString()
+        whenever(mockContext.get(key)).thenReturn(fakeKeyValue)
+
+        // When
+        val result = testedContext.get(key)
+
+        // Then
+        assertThat(result).isEqualTo(fakeKeyValue)
+    }
+
+    // endregion
+
+    // region with key and value
+
+    @Test
+    fun `M return OtelContext W with { key is OTEL_CONTEXT_SPAN_KEY }`() {
+        // Given
+        val key = ContextKey.named<Span>(OtelContext.OTEL_CONTEXT_SPAN_KEY)
+        val value = mock<Span>()
+
+        // When
+        val result = testedContext.with(key, value)
+
+        // Then
+        assertThat(result).isInstanceOf(OtelContext::class.java)
+    }
+
+    @Test
+    fun `M return OtelContext W with { key is OTEL_CONTEXT_ROOT_SPAN_KEY }`() {
+        // Given
+        val key = ContextKey.named<Span>(OtelContext.OTEL_CONTEXT_ROOT_SPAN_KEY)
+        val value = mock<Span>()
+
+        // When
+        val result = testedContext.with(key, value)
+
+        // Then
+        assertThat(result).isInstanceOf(OtelContext::class.java)
+    }
+
+    @Test
+    fun `M return OtelContext W with { key is not recognized }`(forge: Forge) {
+        // Given
+        val fakeKeyName = forge.anAlphabeticalString()
+        val key = ContextKey.named<String>(fakeKeyName)
+        val fakeKeyValue = forge.anAlphabeticalString()
+        val value = fakeKeyValue
+        val wrappedContext: Context = mock()
+        whenever(mockContext.with(key, value)).thenReturn(wrappedContext)
+
+        // When
+        val result = testedContext.with(key, value)
+
+        // Then
+        assertThat(result).isInstanceOf(OtelContext::class.java)
+    }
+
+    @Test
+    fun `M return OtelContext W with consecutively { key is OTEL_CONTEXT_SPAN_KEY }`(forge: Forge) {
```

Comment:
I don't see any key named `OTEL_CONTEXT_SPAN_KEY` in this test

Reply:
yes, bad function name